lib/allure_report_publisher/lib/providers/gitlab.rb

```diff
@@ -62,24 +62,55 @@ def update_pr_description
       #
       # @return [void]
       def add_comment
-        unless comment
+        client.delete_merge_request_discussion_note(project, mr_iid, discussion.id, alert_comment.id) if alert_comment
+
+        if discussion
+          log_debug("Updating summary in comment with id #{discussion.id} in mr !#{mr_iid}")
+          client.update_merge_request_discussion_note(project, mr_iid, discussion.id, main_comment.id,
+                                                      body: url_section_builder.comment_body(main_comment.body))
+        else
           log_debug("Creating comment with summary for mr ! #{mr_iid}")
-          return client.create_merge_request_comment(project, mr_iid, url_section_builder.comment_body)
+          client.create_merge_request_comment(project, mr_iid, url_section_builder.comment_body)
         end
 
-        log_debug("Updating summary in comment with id #{comment.id} in mr !#{mr_iid}")
-        client.edit_merge_request_note(project, mr_iid, comment.id, url_section_builder.comment_body(comment.body))
+        return unless url_section_builder.summary_has_failures?
```

Comment:
I just realised this will not work correctly in context of having multiple different types of jobs reported in single comment.

Basically if one type of test fails and then next one passes, it will resolve the discussion even though there are failures present still

Reply:
Hmmm yeh. Good catch. 

I guess we can fetch the updated comment each time to see if there is an ❌ in the body before removing the alert comment.

Reply:
Yep, we will probably have to parse the whole comment each time. Not ideal, but we fetch it anyway to edit, so it's not too bad.